Sming/Components/rboot/component.mk

```diff
@@ -39,19 +39,22 @@ endif
 
 CONFIG_VARS				+= RBOOT_ROM0_ADDR RBOOT_ROM1_ADDR RBOOT_ROM2_ADDR
 
-# Loation of first ROM (default is sector 2 after rboot and rboot config sector)
+# Location and size of first ROM (default is sector 2 after rboot and rboot config sector)
 RBOOT_ROM0_ADDR			?= 0x002000
+RBOOT_ROM0_SIZE			?= (0x100000 - $(RBOOT_ROM0_ADDR))
 
 # The parameter below specifies the location of the second ROM.
 # You need a second slot for any kind of firmware update mechanism.
 # Leave empty if you don't need a second ROM slot.
 RBOOT_ROM1_ADDR			?=
+RBOOT_ROM1_SIZE			?= (0x100000 - $(RBOOT_ROM1_ADDR))
 
 # The parameter below specifies the location of the GPIO ROM.
 # It is only used when RBOOT_GPIO_ENABLED = 1
 # Note that setting this parameter will only configure rboot.
 # The Sming build system does not create a ROM for this slot.
 RBOOT_ROM2_ADDR			?=
+RBOOT_ROM2_SIZE			?= (0x100000 - $(RBOOT_ROM2_ADDR))
```

Comment:
This size is not used in the build system, because the build system never generates an image for slot2 (the address is only used for configuring the bootloader).
If there is no sanity-checking the memory layout (i.e. check if slots do not overlap with each other or reserved flash areas), `RBOOT_ROM2_SIZE` should be removed IMO.